src/lc.erl

```diff
@@ -0,0 +1,67 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(lc).
+
+-include("lc.hrl").
+
+-export([ is_overloaded/0
+        , join/1
```

Comment:
Perhaps add an API for waiting until the system cools down? So the processes can voluntary suspend themselves. It could be implemented very easily by monitoring the flag process.

Reply:
Good idea, make it happen in 
https://github.com/qzhuyan/lc/pull/1/commits/1721cbed3b53313e95c7ec4d385cfc93e42f605b